dropwizard/src/main/java/com/linecorp/armeria/server/dropwizard/connector/proxy/ArmeriaHttpProxyConnectorFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.dropwizard.connector.proxy;
+
+import java.security.cert.CertificateException;
+import java.util.Objects;
+
+import javax.net.ssl.SSLException;
+import javax.validation.Valid;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.dropwizard.connector.ArmeriaHttpConnectorFactory;
+import com.linecorp.armeria.server.dropwizard.connector.ArmeriaServerDecorator;
+
+import io.dropwizard.jetty.ConnectorFactory;
+
+@JsonTypeName(ArmeriaHttpProxyConnectorFactory.TYPE)
+public class ArmeriaHttpProxyConnectorFactory extends ArmeriaProxyConnectorFactory {
+
+    public static final String TYPE = "armeria-http-proxy";
+    // TODO: This is not configurable
+    @JsonIgnore
+    private final ArmeriaServerDecorator innerFactory;
```

Comment:
Is a single SessionProtocol of PROXY not valid? I saw an error that HTTP or HTTPS needed to be included when I was doing `ServerBuilder.build()`

That being said, some refactoring would have to be done between the non-proxy connectors and these to handle duplicate ServerBuilder code 

One way I can think of this right now is to have a non-Proxy `ConnectorFactory` within the `ProxyConnectorFactory` 

Otherwise, PROXY support seems like a secondary feature to this whole PR

Reply:
> Is a single SessionProtocol of PROXY not valid? I saw an error that HTTP or HTTPS needed to be included when I was doing `ServerBuilder.build()`

Correct. PROXY can’t do anything without HTTP or HTTPS. I think it’s fine as long as 1) a user is allowed to specify PROXY and HTTP(S) connectors at the same port number and 2) the server fails to start up when only PROXY connector is specified.

If you are not familiar with Armeria’s multi-port configuration, we can skip PROXY support for now and consider it later.

```yaml

```

Reply:
Leaving the Abstract proxy factory. Removed the others.